beacon-light-client/plonky2/circuits_executables/bin/balance_verification_circuit_data_generation.rs

```diff
@@ -73,10 +77,9 @@ pub async fn async_main() -> Result<()> {
     }
 
     let mut prev_circuit_data = first_level_data;
-
-    for i in 1..38 {
+    let max_level = if level == None {37} else {clamp(level.unwrap(),1,37)};
```

Comment:
No need for clamp. Just use unwrap_or(37)

Reply:
The clamp is in case a value greater than 37 is supplied

Reply:
If the value is greater than 37, the program returns a few lines above, so the clamp is obsolete